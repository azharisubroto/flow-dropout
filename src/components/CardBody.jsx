import PropTypes from 'prop-types'
const CardBody = ({ className, children, py, ...other }) => {
  return (
    <>
      <div className={`ra-card-body ${className ? className : ''}`} {...other}>
        {children}
      </div>

      {/* Styling */}
      <style jsx>{`
        .ra-card-body {
          padding: 40px;
          background: #fff;
          ${py ? `padding-top: ${py}px;` : ''}
          ${py ? `padding-bottom: ${py}px;` : ''}
        }
      `}</style>
    </>
  )
}

CardBody.propTypes = {
  children: PropTypes.node,
  className: PropTypes.any,
  py: PropTypes.string
}

export default CardBody
