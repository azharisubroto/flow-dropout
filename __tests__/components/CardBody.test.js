import React from 'react'
import { shallow } from 'enzyme'
import CardBody from '@/components/CardBody'

describe('<CardBody />', () => {
  let wrapper, CardBodyComp

  beforeEach(() => {
    CardBodyComp = <CardBody>My CardBody</CardBody>

    wrapper = shallow(CardBodyComp)
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain contain correct text', () => {
    expect(wrapper.find('.ra-card-body').text()).toEqual('My CardBody')
  })
})
