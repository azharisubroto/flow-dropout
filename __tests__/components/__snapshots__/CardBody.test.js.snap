// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBody /> should renders correctly 1`] = `ShallowWrapper {}`;
